src/Courant/Courant_presets.jl

```diff
@@ -0,0 +1,54 @@
+module Courant
+
+
+export Advective_CFL, Diffusive_CFL
+
+using LinearAlgebra, StaticArrays
+using ..VariableTemplates
+using ..MoistThermodynamics
+using ..PlanetParameters
+import ..MoistThermodynamics: internal_energy
+using ..SubgridScaleParameters
+using GPUifyLoops
+using ..MPIStateArrays: MPIStateArray
+using CLIMA.Atmos: AtmosModel, soundspeed
+import CLIMA.DGmethods: BalanceLaw, vars_aux, vars_state,
+                        vars_diffusive
+using ..Mesh.Topologies
+using ..Mesh.Grids
+using ..Mesh.Grids: VerticalDirection, HorizontalDirection, EveryDirection
+using Logging
+using Printf
+using LinearAlgebra
+using CLIMA.Atmos
+
+
+
+function Advective_CFL(m::AtmosModel, state::Vars, aux::Vars,
+                               diffusive::Vars, Δx, Δt, direction=VerticalDirection())
+  k̂ = aux.orientation.∇Φ / norm(aux.orientation.∇Φ)
+  if direction isa VerticalDirection
+    u = dot(state.ρu/state.ρ, k̂) * k̂
+          return Δt * (norm(u) + soundspeed(m.moisture, m.orientation, state,aux)) / Δx
+  elseif direction isa HorizontalDirection
+    u = cross(k̂, cross(state.ρu/state.ρ,k̂))      
+    return Δt * (norm(u) + soundspeed(m.moisture, m.orientation, state,
+                                            aux)) / Δx
+  end
+end
+
+function Diffusive_CFL(m::AtmosModel, state::Vars, aux::Vars,
+                               diffusive::Vars, Δx, Δt, direction=VerticalDirection())
+  ν, τ, sdiff = turbulence_tensors(m.turbulence, state, diffusive, aux, 0)
```

Comment:
Can't we pass the value of the eddy viscosity which is already computed elsewhere instead of re-computing it?

Reply:
I'm looking into seeing if I can do this with update-diffusive.

Reply:
Note that there is no guarantee that the current `diffusive` and `aux` arrays is consistent with the `state` outside of the dg model [1], so for safety reason you may want to recompute these updates before calculating the CFL.

> Can't we pass the value of the eddy viscosity which is already computed elsewhere instead of re-computing it?

It is never stored anymore, so you must recompute it [2].

[1]: https://github.com/climate-machine/CLIMA/blob/7977f64a67cb745f3f23a6c90c0c4ad12328c915/src/DGmethods/DGmodel.jl#L20-L175
[2]: https://github.com/climate-machine/CLIMA/blob/c117b4c7eb8b33b3133fcd5b4886269d1356bc08/src/Atmos/Model/AtmosModel.jl#L175-L182